### DISCUSSION TITLE
default value for integer (time) field in sql files

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5206] [General]
Hi,
I thought that I made mistake, but you use it similar way  in core
f.e. 
`  news_datestamp int(10) unsigned NOT NULL default '0',`

For some users there is error 
```
Query: ALTER TABLE `fanfiction_authors` ADD `date_tmp` int(11) NOT NULL default '0'
Error: (1067) Invalid default value for 'date'
```
problem is in default value.  
Is this somehow related to the database version? For somebody it works, for somebody doesn't.

It is change from original string date value to integer with UNIX_TIMESTAMP().

 Thanks for help.

**Comment by Vodhin** [comment]
I do not think you need "default '0'" since the field is an integer.

**Reply by Jimmi08** [reply]
The problem was that there were data that after conversion has value 0 - but database was run with no-zero mode for datetime . Something like that.
default '0'  I took from core SQL file, all integer fields are done this way. 
Thanks for trying to help.
